### PR TITLE
feat: Surface code that gets uploaded in "Report FP" panel [ROAD-551]

### PR DESCRIPTION
**Pull request by michelkaporin**
screen

Screenshot 2022-01-31 at 08 34 33

### CODE REVIEW

File: src/snyk/snykCode/falsePositive/falsePositive.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  /**
+   * May throw an error if file cannot be resolved.
+   */
+  async getContent(): Promise<string> {
+    let content = this.getMainHeader();
+
+    for await (const file of this.files) {
+      const doc = await this.workspace.openFileTextDocument(file);
+      content += this.appendFileHeader(doc.getText(), file);
```
Comment (bastiandoetsch) [inline src/snyk/snykCode/falsePositive/falsePositive.ts:40]: Any chance to set the type here to get syntax highlighting?

File: src/snyk/common/commands/commandController.ts [hunk 2]
```diff
@@ -106,6 +110,35 @@ export class CommandController {
...
+  async reportFalsePositive(arg: ReportFalsePositiveCommandArg): Promise<void> {
+    const suggestion = arg.suggestion;
+    if (!suggestion.markers || suggestion.markers.length === 0) {
+      return;
+    }
+
+    let content = '';
+    try {
+      const falsePositive = new FalsePositive(this.workspace, suggestion.uri, suggestion.markers);
+      content = await falsePositive.getContent();
```
[the remark this answers is not in the document]
Comment (michelkaporin) [reply src/snyk/common/commands/commandController.ts:122]: It is needed to obtain the result from the promise that resolves asynchronously. Marking function async allows awaiting inside function body.

File: src/snyk/snykCode/falsePositive/falsePositive.ts (hunk 1 above)
Comment (michelkaporin) [reply src/snyk/snykCode/falsePositive/falsePositive.ts:40]: It actually infers the type as a string since we set it to this type on line 36.

**Review by bastiandoetsch — APPROVED** [review]
Hm... did I overlook the tests? Didn't find any issues glaring at me besides I don't think I saw new tests.

**Comment by michelkaporin** [comment]
@bastiandoetsch Fair point, I've left the tests out of this PR, since most of the logic is UI-related and it would require integration tests to cover it properly - which doesn't work at the moment. I'll test FalsePositive class in an upcoming PR though 👍

**Pull request merged**